plugins/catalog-backend/src/ingestion/processors/ApiDefinitionAnnotationProcessor.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ComponentEntity,
+  Entity,
+  LocationSpec,
+} from '@backstage/catalog-model';
+import { LocationProcessor, LocationProcessorEmit } from './types';
+
+const API_DEFINITION_TYPE_ANNOTATION = /^backstage.io\/api-definition-(.+)$/;
+
+export function extractApiDefinitionTypeAnnotations(entity: Entity): string[] {
+  if (!entity.metadata.annotations) {
+    return [];
+  }
+
+  return Object.keys(entity.metadata.annotations).filter(a =>
+    a.match(API_DEFINITION_TYPE_ANNOTATION),
+  );
+}
+
+export function extractApiDefinitionType(annotationName: string): string {
+  return annotationName.match(API_DEFINITION_TYPE_ANNOTATION)![1];
+}
+
+export class ApiDefinitionAnnotationProcessor implements LocationProcessor {
+  async processEntity(
+    entity: Entity,
+    location: LocationSpec,
+    emit: LocationProcessorEmit,
+  ): Promise<Entity> {
+    // TODO: Case insensitive?
```

Comment:
I guess, yeah... Everything else tries to be, internally, but then again we kinda expect users to write it exactly like that in yaml.